### DISCUSSION TITLE
Why is my supabase API call not returning?

### DISCUSSION THREAD

**Discussion by GaryAustin1** [supabase/supabase#19058] [Troubleshooting]
_This is a copy of a troubleshooting article on Supabase's docs site. It may be missing some details from the original. View the [original article](https://supabase.com/docs/guides/troubleshooting/why-is-my-supabase-api-call-not-returning-PGzXw0)._

There is currently a bug in supabase-js which results in a deadlock if any async API call is made in `onAuthStateChange` code. If a call is made in the handler then the next Supabase call anywhere using that client will hang and not return.

For now a workaround is needed until this is fixed.

<https://github.com/supabase/gotrue-js/issues/762#issuecomment-1780006492>

**Comment by hf** [comment]
This behavior and best practices are documented in [`supabase.auth.onAuthStateChange` reference docs](https://supabase.com/docs/reference/javascript/auth-onauthstatechange).
